pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -53,7 +53,7 @@ in
 , maintainers ? []
 , doCoverage ? false
 , doHaddock ? !(ghc.isHaLVM or false)
-, doHaddockInterfaces ? true
+, doHaddockInterfaces ? stdenv.lib.versionAtLeast ghc.version "8.12"
```

Comment:
What does ghc-8.12.x have to do with this PR?

Reply:
8.12.x will turn Haddock errors into warnings, provided [!2377](https://gitlab.haskell.org/ghc/ghc/-/merge_requests/2377) lands before the cut